includes/functions.php

```diff
@@ -2058,39 +2060,104 @@ function tsml_scan_data_source($data_source_url)
-		$meetings = tsml_import_reformat($meetings);
-	}*/
+	/* filter out all but the data source meetings  */
+	foreach ($all_db_meetings as $db_meeting) {
+	    $db_id = $db_meeting['id'];
+		if (in_array($db_id, $ds_ids)) {
+		    array_push($db_meetings, $db_meeting);
+		}
+	}
+
+	// create array of db slugs for matching
+	$db_slugs = array_column($db_meetings, 'slug', 'id');
+	sort($db_slugs);
+	$db_slugs_cnt = count($db_slugs);
 
-	if (count($meetings) !== $data_source_count_meetings) {
+	// list changed and new meetings found in the ds feed
+	foreach ($meetings as $meeting) {
 
-		if (count($meetings) > $data_source_count_meetings) {
-			$meetings_updated[] = 'Feed Records: ' . count($meetings) . ' more than DB Records: ' . $data_source_count_meetings;
-		} elseif (count($meetings) < $data_source_count_meetings) {
-			$meetings_updated[] = 'Feed Records: ' . count($meetings) . ' less than DB Records: ' . $data_source_count_meetings;
+	    $mtg_slug =  $meeting['slug'];
+		if (is_array($meeting['day'])) {
+			$dow = implode("",  $meeting['day']);
 		} else {
-			$meetings_updated[] = 'Record count different';
+			$dow = $meeting['day'];
+		}
+		
+		if ( is_numeric( $mtg_slug )) {
+			$mtg_slug .= '-' . $dow; 
 		}
-	}
 
-	foreach ($meetings as $meeting) {
+		// match feed/database on unique slug
+	    $is_matched = in_array( $mtg_slug, $db_slugs );
+
+		// add slug to feed array to help determine current db removals later on...
+		$feed_slugs[$meeting['slug']] = $mtg_slug;
 
 		// has meeting been updated?
 		$updated = $meeting['updated'];
 		$cur_meeting_lastupdate = strtotime($updated);
-
 		if ($cur_meeting_lastupdate > $data_source_last_refresh) {
+			$is_there_change = true;
 			$mtg_name = $meeting['name'];
-			$mtg_updte = date("l F j, Y  h:i a", $cur_meeting_lastupdate);
-			$meetings_updated[] = "$mtg_name updated: $mtg_updte";
+			$mtg_updte = date("M j, Y  g:i a", $cur_meeting_lastupdate);
+			if ( is_numeric( $dow )) {
+				$dayofweek = $days[date($dow)]; 
+			} else {
+				$dayofweek = $dow; 
+			}
+			
+			if ($is_matched) {
+				$message_lines[] = "<tr style='color:gray;'><td>Change</td><td >$mtg_name</td><td>$dayofweek</td><td>$mtg_updte</td></tr>";
+			}
+			else {
+				$message_lines[] = "<tr style='color:green;'><td>Add New</td><td >$mtg_name</td><td>$dayofweek</td><td>$mtg_updte</td></tr>";
+			}
+		}
+	}
+
+	sort($feed_slugs);
+
+	// list removed meetings found in local db
+	foreach ($db_meetings as $db_meeting) {
+
+	    $mtg_slug =  $db_meeting['slug'];
+		if (is_array($db_meeting['day'])) {
+			$dow = implode("",  $db_meeting['day']);
+		} else {
+			$dow = $db_meeting['day'];
 		}
+
+	    $is_matched = in_array( $mtg_slug, $feed_slugs );
+
+		if (!$is_matched) {
+		    $is_there_change = true;
+			if ( is_numeric( $dow )) {
+				$dayofweek = $days[date($dow)]; 
+			} else {
+				$dayofweek = $dow; 
+			}
+			$updated = $data_source_last_refresh;
+			$cur_meeting_lastupdate = strtotime($updated);
```

Comment:
i would love it if we could be a little more intentional about variable names. i think that aside from obvious shortenings like `db` or `id` that we should try to spell out words. also snake casing should be consistently applied between words. so this could be `$current_meeting_last_update` and `$mtg_updte` could be `$meeting_update`.

this will make for a cleaner appearance and will be much easier for developers who don't have english as a native language.

Reply:
Will do...